pkg/tools/mcp_tool.go

```diff
@@ -0,0 +1,301 @@
+}
+
+// convertToBoolean attempts to convert value to a boolean
+func convertToBoolean(value any) any {
+	switch v := value.(type) {
+	case string:
+		if b, err := strconv.ParseBool(v); err == nil {
+			return b
+		}
+	case bool:
+		return v
+	case int:
+		return v != 0
+	}
+	return value // Keep original if conversion fails
+}
+
+// =============================================================================
+// MCP Integration Functions
+// =============================================================================
+
+// registerToolsFromConnectedServers discovers tools from all connected MCP servers and registers them
+func registerToolsFromConnectedServers(ctx context.Context) error {
+	if mcpManager == nil {
+		return nil
+	}
+
+	// Discover tools from connected servers
+	serverTools, err := mcpManager.RefreshToolDiscovery(ctx)
+	if err != nil {
+		return err
+	}
+
+	toolCount := 0
+	for serverName, tools := range serverTools {
+		for _, toolInfo := range tools {
+			// Create schema for the tool
+			schema, err := convertToolToGollm(&mcp.Tool{
+				Name:        toolInfo.Name,
+				Description: toolInfo.Description,
+			})
+			if err != nil {
+				klog.Warningf("Failed to convert schema for tool %s from server %s: %v", toolInfo.Name, serverName, err)
+				continue
+			}
+
+			// Create MCP tool wrapper
+			mcpTool := NewMCPTool(serverName, toolInfo.Name, toolInfo.Description, schema, mcpManager)
+
+			// Register with the tools system
+			RegisterTool(mcpTool)
+			toolCount++
+		}
+	}
+
+	if toolCount > 0 {
+		klog.InfoS("Registered MCP tools", "totalTools", toolCount)
+	}
+
+	return nil
+}
+
+// InitializeMCPClient explicitly initializes MCP client functionality when --mcp-client flag is used
+func InitializeMCPClient() error {
+	// Initialize the MCP manager using the new pkg/mcp functions
+	manager, err := mcp.InitializeManager()
+	if err != nil {
+		return err
+	}
+
+	mcpManager = manager
+
+	// Start connection and tool discovery synchronously to ensure tools are available before conversation starts
+	ctx := context.Background()
+
+	// Connect to servers
+	if err := mcpManager.DiscoverAndConnectServers(ctx); err != nil {
+		klog.V(2).Info("MCP server connection failed", "error", err)
+		return fmt.Errorf("MCP server connection failed: %w", err)
+	}
+
+	// Register discovered tools
+	if err := registerToolsFromConnectedServers(ctx); err != nil {
+		klog.V(2).Info("MCP tool registration failed", "error", err)
+		return fmt.Errorf("MCP tool registration failed: %w", err)
+	}
+
+	return nil
+}
+
+// GetMCPManager returns the global MCP manager instance (for UI integration)
+func GetMCPManager() *mcp.Manager {
+	return mcpManager
+}
+
+// RefreshMCPTools forces a refresh of MCP tools (for manual refresh scenarios)
+func RefreshMCPTools(ctx context.Context) error {
+	if mcpManager == nil {
+		return fmt.Errorf("MCP manager not initialized")
+	}
+
+	return registerToolsFromConnectedServers(ctx)
+}
```

Comment:
All these can become methods on the manager instance and we can hold the instance as dependency in main.go during bootstrap.